packages/smooth_app/lib/pages/product/product_page/header/product_tabs.dart

```diff
@@ -0,0 +1,239 @@
+import 'package:smooth_app/pages/prices/prices_card.dart';
+import 'package:smooth_app/pages/product/product_page/header/reorder_bottom_sheet.dart';
+import 'package:smooth_app/pages/product/website_card.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_app/themes/smooth_theme_colors.dart';
+
+class ProductTab {
+  ProductTab({
+    required this.labelBuilder,
+    required this.builder,
+  });
+
+  final Widget Function(BuildContext) labelBuilder;
+  final Widget Function(Product) builder;
+}
+
+class ProductTabBar extends StatelessWidget {
+  const ProductTabBar({
+    required this.tabController,
+    required this.tabs,
+  });
+
+  final TabController tabController;
+  final List<ProductTab> tabs;
+
+  @override
+  Widget build(BuildContext context) {
+    final SmoothColorsThemeExtension themeExtension =
+        context.extension<SmoothColorsThemeExtension>();
+
+    return SliverPersistentHeader(
+      delegate: TabBarDelegate(
+        TabBar(
+          controller: tabController,
+          isScrollable: true,
+          tabs: tabs
+              .map((ProductTab tab) => Tab(
+                    child: tab.labelBuilder(context),
+                  ))
+              .toList(growable: false),
+          labelColor: Theme.of(context).primaryColor,
+          labelStyle: const TextStyle(
+            fontWeight: FontWeight.bold,
+            fontSize: 15.0,
+          ),
+          unselectedLabelStyle: const TextStyle(
+            fontWeight: FontWeight.bold,
+            fontSize: 15.0,
+          ),
+          overlayColor: WidgetStateProperty.all(
+            themeExtension.primaryLight,
+          ),
+          dividerColor: themeExtension.primaryNormal,
+          automaticIndicatorColorAdjustment: false,
+          indicatorSize: TabBarIndicatorSize.tab,
+          indicator: BoxDecoration(
+            border: Border(
+              bottom: BorderSide(
+                color: Theme.of(context).primaryColor,
+                width: 3.0,
+              ),
+            ),
+            color: themeExtension.primaryLight,
+          ),
+          tabAlignment: TabAlignment.start,
+        ),
+      ),
+      pinned: true,
+    );
+  }
+
+  static List<ProductTab> extractTabsFromProduct({
+    required BuildContext context,
+    required Product product,
+  }) {
+    final List<ProductTab> tabs = <ProductTab>[];
+
+    final List<KnowledgePanelElement> roots =
+        KnowledgePanelsBuilder.getRootPanelElements(product);
+    for (final KnowledgePanelElement root in roots) {
+      final List<Widget> children = KnowledgePanelsBuilder.getChildren(
+        context,
+        panelElement: root,
+        product: product,
+        onboardingMode: false,
+      );
+
+      tabs.add(
+        ProductTab(
+          labelBuilder: (BuildContext c) =>
+              Text((children.first as KnowledgePanelTitle).title),
+          builder: (Product p) => ListView(
```

Comment:
ListView is a shitty Widget.
If you have more than a few items, please use the `.builder` variant